### PR TITLE
Set up for the DB

### PR DESCRIPTION
**Pull request by NicholasDanks**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -4,4 +4,18 @@ ruby '2.2.3'
 gem 'sinatra'
 gem 'thin'
 gem 'json'
+gem 'sequel'
+gem 'sqlite3'
```
Comment (rjollet) [inline Gemfile]: sqlite3 is for developement and test but not for production so you can put 

```
group :development, :test do
  gem 'sqlite3'
end
```

**Comment by NicholasDanks** [comment]
Done

File: db/migrations/001_trackers_create.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+require 'sequel'
+
+Sequel.migration do 	
+  change do
+  	create_table(:trackers) do
+  	  primary_key :id
+  	  foreign_key :campaign_id
+  	  
+  	  String :label, unique: true, null: false
```
Comment (rjollet) [inline db/migrations/001_trackers_create.rb]: label is not unique but the couple (campaign_id, label)

```
unique [:campaign_id, label]
```

File: db/migrations/001_trackers_create.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  	  String :url, unique: true, null: false
+  	end
+  end
+end
```
Comment (rjollet) [inline db/migrations/001_trackers_create.rb]: you need to put an empty line at this end of your files there is this issue in all your migrations files

File: db/migrations/001_trackers_create.rb (hunk 2 above)
Comment (NicholasDanks) [inline db/migrations/001_trackers_create.rb]: Done

File: db/migrations/002_campaign_create.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+require 'sequel'
```
Comment (rjollet) [inline db/migrations/002_campaign_create.rb]: files name should be 002_campaigns_create.rb to be consistent

File: Gemfile [hunk 5]
```diff
@@ -5,12 +5,13 @@ gem 'sinatra'
...
 gem 'sequel'
-gem 'sqlite3'
+
 
 group :development do
   gem 'rerun'
   gem 'tux'
-  gem 'hirb'	
+  gem 'hirb'
+  gem 'sqlite3'
```
Comment (rjollet) [inline Gemfile]: you need to create a group for both test and development sqlite3 is also needed for test

```
group :development, :test do
  gem 'sqlite3'
end
```

**Comment by NicholasDanks** [comment]
Ok guys. I completed the models. It's a little sketchy, but a good base I think. Please can you check it and agree pull.

File: models/campaign.rb [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+# Holds the visits information for each pixel tracker
+class Visit < Sequel::Model
+  one_to_many :trackers
+
+  def to_json(options = {})
+  	JSON({ type: 'campaign',
+  	       id: id,
+  	       label: label,
+  	  },
+  		options)
```
Comment (rjollet) [inline models/campaign.rb:13]: you mixed white-space and tab please convert tab into white-spaces

File: models/visit.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
+require 'json'
+require 'sequel'
+
+# Holds the visits information for each pixel tracker
+class Visit < Sequel::Model
+  many_to_one :trackers
+
+  def to_json(options = {})
+  	JSON({ type: 'tracker',
```
Comment (rjollet) [inline models/visit.rb]: type view

**Pull request merged**